Foundation/NSData.swift

```diff
@@ -104,21 +104,21 @@ public class NSData : NSObject, NSCopying, NSMutableCopying, NSSecureCoding {
     }
     
     deinit {
-        if _bytes != nil {
-            _deallocHandler?.handler(_bytes, _length)
+        if let allocatedBytes = _bytes {
+            _deallocHandler?.handler(allocatedBytes, _length)
         }
         if self.dynamicType === NSData.self || self.dynamicType === NSMutableData.self {
             _CFDeinit(self._cfObject)
         }
     }
     
-    internal init(bytes: UnsafeMutablePointer<Void>, length: Int, copy: Bool, deallocator: ((UnsafeMutablePointer<Void>, Int) -> Void)?) {
+    internal init(bytes: UnsafeMutablePointer<Void>?, length: Int, copy: Bool, deallocator: ((UnsafeMutablePointer<Void>, Int) -> Void)?) {
         super.init()
         let options : CFOptionFlags = (self.dynamicType == NSMutableData.self) ? __kCFMutable | __kCFGrowable : 0x0
         if copy {
             _CFDataInit(unsafeBitCast(self, to: CFMutableData.self), options, length, UnsafeMutablePointer<UInt8>(bytes), length, false)
-            if let handler = deallocator {
-                handler(bytes, length)
+            if let handler = deallocator, allocatedBytes = bytes {
```

Comment:
nope, that is incorrect behavior, if the passed pointer to the deallocator is null it is perfectly valid in the objc sense